AutoGrade/views.py

```diff
@@ -478,3 +487,40 @@ def logout_student(request):
     else:
         logout(request)
         return redirect('login')
+
+
+@login_required(login_url='login')
+def request_extension(request):
+    student = Student.objects.get(user = request.user)
+    assignment_id = request.GET.get('aid')
+
+
+
+    selected_assignment = Assignment.objects.get(id=assignment_id)
```

Comment:
@recluze 

I think we should check ObjectDoesntExist exception here  

`try: 
    student = Student.objects.get(user = request.user) # If admin tries to acces this page. Rare.
    assignment_id = request.GET.get('aid')
    selected_assignment = Assignment.objects.get(id=assignment_id)
except ObjectDoesNotExist:
    logging.warn("Extension (You are logged): " + str(student) + " on " + str(assignment_id) + " - " + str(selected_assignment))
    out = "You are logged."
    return HttpResponse(out, content_type="text/plain", status=status)
`

Reply:
Just checked. When logged in as admin, `request.user` does give back the same user. So, can process the request as usual.

Reply:
I mean logged in through admin account and then "login as student". Request works. Of course, admins shouldn't do it but will never do so anyway since they have the backend for it.

Reply:
I was away for the university trip (Naran) and doesn't have the internet to update you.

This get() method throws an exception when no record is available.

**Two scenarios:**
1) This situation occurred with me when I was logged in as `guru` and tried to open ".../request_extension?aid=1" in the browser (That is why its very rare =]). This exception was thrown by `Student.objects.get(...)`.
2) Secondly, `Assignment.objects.get(...)` will throw an exception when someone try to play with `aid` value for example ".../request_extension?aid=`1234`" or ".../request_extension/". 

And I was wondering whether transactions will be useful for this case (request_extension view) or not.